anvil/src/window_map.rs

```diff
@@ -252,6 +271,15 @@ impl WindowMap {
         }
         if let Some((i, surface)) = found {
             let winner = self.windows.remove(i);
+
+            // Take activation away from all the windows
+            for window in self.windows.iter() {
+                window.toplevel.set_activated(false);
+            }
+
+            // Give activation to our winner
+            winner.toplevel.set_activated(true);
+
```

Comment:
Is it really necessary to remove the window (and cause a reorganization of the `Vec`) and insert it back? I think we can just check in the for loop that the current window is not the winner before disabling it.

Reply:
It was not written by me, it is part of bringing surface to top. I just added the activation part in between.
https://github.com/Smithay/smithay/blob/da937d498e33c991d1eebf0de2e96f30fbe90b73/anvil/src/window_map.rs#L255

Reply:
Oh, right, bringing it to the top while preserving the ordering of the rest. I forgot about that part, my bad!